### ISSUE TITLE
Network Protocol Error

### ISSUE THREAD

**Issue by Wikinger8** [emilyploszaj/emi#620]
https://github.com/emilyploszaj/emi/assets/85295145/190baea7-db19-4bcc-85dc-78fc63d2643b

![image](https://github.com/emilyploszaj/emi/assets/85295145/812f63e5-9e0a-4e95-8723-040fda76e46e)
[latest.log](https://github.com/user-attachments/files/16117130/latest.log)
[disconnect-2024-07-06_21.33.51-client.txt](https://github.com/user-attachments/files/16117131/disconnect-2024-07-06_21.33.51-client.txt)

Somehow Sodium is the problem.

**Comment by emilyploszaj** [comment]
Please report this to sodium.

**Comment by Wikinger8** [comment]
![image](https://github.com/emilyploszaj/emi/assets/85295145/5bc88676-1106-4d3c-a68e-da41b17f2e42)
https://github.com/CaffeineMC/sodium-fabric/issues/2589

**Comment by MeeniMc** [comment]
This can be replicated in singleplayer: the game will load to completion, and sometimes crash when first opening the inventory. 

Log contains the following error, which does not crash the game
```
[13:51:05] [Render thread/INFO] (EMI) [EMI] Tags synchronized, reloading EMI
[13:51:05] [Render thread/INFO] (EMI) [EMI] Starting Reload
[13:51:05] [Thread-17/INFO] (EMI) [EMI] Starting EMI reload...
[13:51:05] [Thread-17/INFO] (EMI) [EMI] Clearing data
[13:51:05] [Thread-17/INFO] (EMI) [EMI] Initializing plugin from emi
[13:51:05] [Thread-17/INFO] (EMI) [EMI] Initialized plugin from emi in 6ms
[13:51:05] [Thread-17/INFO] (EMI) [EMI] Initializing plugin from owo
[13:51:05] [Thread-17/INFO] (EMI) [EMI] Initialized plugin from owo in 0ms
[13:51:05] [Thread-17/INFO] (EMI) [EMI] Processing tags
[13:51:05] [Thread-17/INFO] (EMI) [EMI] Constructing index
[13:51:05] [Thread-17/ERROR] (EMI) [EMI] Creative item group Â§fColored Blocks threw while EMI was attempting to construct the index, items may be missing.
[13:51:05] [Thread-17/INFO] (Minecraft) [STDERR]: java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 513
[13:51:05] [Thread-17/INFO] (Minecraft) [STDERR]: 	at it.unimi.dsi.fastutil.objects.ObjectLinkedOpenCustomHashSet.add(ObjectLinkedOpenCustomHashSet.java:340)
[13:51:05] [Thread-17/INFO] (Minecraft) [STDERR]: 	at java.base/java.util.AbstractCollection.addAll(AbstractCollection.java:338)
[13:51:05] [Thread-17/INFO] (Minecraft) [STDERR]: 	at it.unimi.dsi.fastutil.objects.ObjectLinkedOpenCustomHashSet.addAll(ObjectLinkedOpenCustomHashSet.java:314)
[13:51:05] [Thread-17/INFO] (Minecraft) [STDERR]: 	at net.minecraft.class_1761.handler$cnn000$fabric-item-group-api-v1$getStacks(class_1761.java:1587)
[13:51:05] [Thread-17/INFO] (Minecraft) [STDERR]: 	at net.minecraft.class_1761.method_47306(class_1761.java:108)
[13:51:05] [Thread-17/INFO] (Minecraft) [STDERR]: 	at dev.emi.emi.registry.EmiStackList.reload(EmiStackList.java:83)
[13:51:05] [Thread-17/INFO] (Minecraft) [STDERR]: 	at dev.emi.emi.runtime.EmiReloadManager$ReloadWorker.run(EmiReloadManager.java:175)
[13:51:05] [Thread-17/INFO] (Minecraft) [STDERR]: 	at java.base/java.lang.Thread.run(Thread.java:1583)
[13:51:05] [Thread-17/ERROR] (EMI) [EMI] java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 513
	at it.unimi.dsi.fastutil.objects.ObjectLinkedOpenCustomHashSet.add(ObjectLinkedOpenCustomHashSet.java:340)
	at java.base
[13:51:05] [Thread-17/ERROR] (EMI) [EMI] java.util.AbstractCollection.addAll(AbstractCollection.java:338)
	at it.unimi.dsi.fastutil.objects.ObjectLinkedOpenCustomHashSet.addAll(ObjectLinkedOpenCustomHashSet.java:314)
	at net.minecraft.class_1761.handler$cnn000$fabric-item-group-api-v1$getStacks(class_1761.java:1587)
	at net.minecraft.class_1761.method_47306(class_1761.java:108)
	at dev.emi.emi.registry.EmiStackList.reload(EmiStackList.java:83)
	at dev.emi.emi.runtime.EmiReloadManager$ReloadWorker.run(EmiReloadManager.java:175)
	at java.base
[13:51:05] [Thread-17/ERROR] (EMI) [EMI] java.lang.Thread.run(Thread.java:1583)
```

Later, when first opening the (creative, not sure if relevant) inventory , this crashes the game 
```
[13:51:08] [Render thread/ERROR] (Minecraft) Unreported exception thrown!
java.lang.IllegalStateException: Item group has no page
	at net.minecraft.class_1761.fabric_getPage(class_1761.java:1596) ~[client-intermediary.jar:?]
	at net.minecraft.class_481.getPage(class_481.java:2214) ~[client-intermediary.jar:?]
	at net.minecraft.class_481.isGroupVisible(class_481.java:2204) ~[client-intermediary.jar:?]
	at net.minecraft.class_481.handler$cnp000$fabric-item-group-api-v1$setSelectedTab(class_481.java:2176) ~[client-intermediary.jar:?]
	at net.minecraft.class_481.method_2466(class_481.java) ~[client-intermediary.jar:?]
	at net.minecraft.class_481.method_25426(class_481.java:439) ~[client-intermediary.jar:?]
	at net.minecraft.class_437.method_25423(class_437.java:309) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1507(class_310.java:1143) ~[client-intermediary.jar:?]
	at net.minecraft.class_490.method_25426(class_490.java:50) ~[client-intermediary.jar:?]
	at net.minecraft.class_437.method_25423(class_437.java:309) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1507(class_310.java:1143) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1508(class_310.java:2068) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.redirect$dii000$g4mespeed$onTickRedirectHandleInputEvents(class_310.java:20349) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1574(class_310.java:1938) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1246) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:882) [client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:256) [client-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470) [fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.15.11.jar:?]
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:100) [NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129) [NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70) [NewLaunch.jar:?]
```

**Comment by emilyploszaj** [comment]
The related item group exception should be resolved in 1.1.11
